.github/workflows/pull-request-management.yml

```
@@ -683,32 +696,72 @@ jobs:
       - name: Run pytest via tox for ${{ matrix.python }}
         working-directory: python-avd
         run: |
-          tox
-      - name: Upload coverage from pytest
-        # Coverage only runs as part of 3.11.
-        if: |
-          matrix.python == '3.11'
+          tox -- ${{ matrix.suite.pytest_args }}
+      - name: Rename the coverage file
+        if: matrix.python == '3.11'
+        run: |
+          mv ./python-avd/.coverage.* ./coverage-${{ matrix.suite.id }}.dat
+      - name: Upload pytest coverage data
+        if: matrix.python == '3.11'
         uses: actions/upload-artifact@v4
         with:
-          name: pytest-coverage
-          path: python-avd/coverage.xml
-
+          name: coverage-data-${{ matrix.suite.id }}
+          path: coverage-${{ matrix.suite.id }}.dat
       - name: Upload eos_designs compiled templates from pytest
-        if: |
-          matrix.python == '3.11'
+        if: matrix.suite.upload_compiled_templates == true && matrix.python == '3.11'
         uses: actions/upload-artifact@v4
         with:
           name: pytest-eos-designs-compiled-templates
           path: python-avd/pyavd/_eos_designs/j2templates/compiled_templates/
-
       - name: Upload eos_cli_config_gen compiled templates from pytest
-        if: |
-          matrix.python == '3.11'
+        if: matrix.suite.upload_compiled_templates == true && matrix.python == '3.11'
         uses: actions/upload-artifact@v4
         with:
           name: pytest-eos-cli-config-gen-compiled-templates
           path: python-avd/pyavd/_eos_cli_config_gen/j2templates/compiled_templates/
 
+  generate_pyavd_coverage_report:
+    name: Test pyavd python 3.11 coverage report
+    runs-on: ubuntu-latest
+    needs: [test_pyavd]
+    steps:
+      - uses: actions/checkout@v4
```

we have bumped the checkout action version recently :)

```suggestion
      - uses: actions/checkout@v5
```